src/main/java/com/powsybl/openloadflow/network/LfContingency.java

```diff
@@ -99,10 +107,18 @@ public double getActivePowerLoss() {
         }
         buses.forEach(b -> branches.addAll(b.getBranches()));
 
+        Set<Pair<LfShunt, Double>> shunts = new HashSet<>(1);
+        for (Pair<String, Double> shuntAndB : propagatedContingency.getShuntIdsToLose()) {
+            LfShunt shunt = network.getShuntById(shuntAndB.getKey());
+            if (shunt != null) {
+                shunts.add(Pair.of(shunt, shuntAndB.getValue()));
```

Comment:
The B value seems to be the global one (bus level) and not the one the targeting shunt?

Reply:
In this case I think there is something tricky to implement in the LfShuntImpl::updateState to find the right section taking into account triggered IIDM shunt

Reply:
Indeed, in this PR, it is not obvious but I don't support shunt with voltage control on, so it means that I will always have fixed shunt and a global B. That is why it is working... I am going to add a unit test.